From 8186cda653aefeced401afbb31fc009446c0e391 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Tue, 11 Aug 2015 15:55:00 +0800
Subject: [PATCH 0371/1691] MLK-11344-2: firmware: imx: sdma: add sdma firmware
 for i.mx6 and i.mx7d

commit 8186cda653aefeced401afbb31fc009446c0e391 from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

add sdma firmware for i.mx6 and i.mx7d

Signed-off-by: Robin Gong <b38343@freescale.com>
---
 firmware/Makefile                     |   1 +
 firmware/imx/sdma/sdma-imx6q.bin.ihex | 139 ++++++++++++++++++++++++++++++++
 firmware/imx/sdma/sdma-imx7d.bin.ihex | 145 ++++++++++++++++++++++++++++++++++
 3 files changed, 285 insertions(+)
 create mode 100644 firmware/imx/sdma/sdma-imx6q.bin.ihex
 create mode 100644 firmware/imx/sdma/sdma-imx7d.bin.ihex

diff --git a/firmware/Makefile b/firmware/Makefile
index e297e1b..ee4c05c 100644
--- a/firmware/Makefile
+++ b/firmware/Makefile
@@ -61,6 +61,7 @@ fw-shipped-$(CONFIG_DRM_RADEON) += radeon/R100_cp.bin radeon/R200_cp.bin \
 				   radeon/RV770_pfp.bin radeon/RV770_me.bin \
 				   radeon/RV730_pfp.bin radeon/RV730_me.bin \
 				   radeon/RV710_pfp.bin radeon/RV710_me.bin
+fw-shipped-$(CONFIG_IMX_SDMA) += imx/sdma/sdma-imx6q.bin imx/sdma/sdma-imx7d.bin
 fw-shipped-$(CONFIG_DVB_AV7110) += av7110/bootcode.bin
 fw-shipped-$(CONFIG_DVB_TTUSB_BUDGET) += ttusb-budget/dspbootcode.bin
 fw-shipped-$(CONFIG_E100) += e100/d101m_ucode.bin e100/d101s_ucode.bin \
diff --git a/firmware/imx/sdma/sdma-imx6q.bin.ihex b/firmware/imx/sdma/sdma-imx6q.bin.ihex
new file mode 100644
index 0000000..5275e15
--- /dev/null
+++ b/firmware/imx/sdma/sdma-imx6q.bin.ihex
@@ -0,0 +1,139 @@
+:1000000053444D4103000000020000001C000000AA
+:1000100029000000C0000000D2070000820200009A
+:10002000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFE0
+:10003000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFD0
+:10004000FFFFFFFFFFFFFFFF3F1A0000FFFFFFFF63
+:10005000EB020000B1180000FFFFFFFF9D1A000037
+:10006000FFFFFFFFC0030000FFFFFFFFFFFFFFFFD9
+:10007000FFFFFFFFAB020000FFFFFFFF7B0300005D
+:10008000FFFFFFFFFFFFFFFF4C0400006E040000B6
+:10009000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF70
+:1000A000FFFFFFFF0018000058180000EB190000C8
+:1000B0004C1B0000E71A0000831B00000018000022
+:1000C000E3C1DB57F352016AFB52D36AFB521C1A9D
+:1000D000C36AE8621102FF3A3008D002337C8F0015
+:1000E000D500017D8D00A005EB5D7804037D7904CA
+:1000F0001C7D207C7904157CEE566006057D096523
+:10010000317E0A622F7E28980A622C7E09652A7E3B
+:1001100012051205AD02C86A257F03200048ED7C58
+:10012000409802780962C86A3F98150002780A620E
+:10013000C86A3F981500150002780B62C86A127CE5
+:10014000DF6D107FFB52041AC36AD36AC862042AA7
+:10015000C86AFB52D36A0000EB55004D077DFAC117
+:10016000DB5706980700CC680C6813C20AC20198D6
+:10017000D9C1E3C1DB57E35FE357F352216A8F0034
+:10018000D500017D8D00A005EB5DFB567804037D55
+:1001900079042A7D317C7904207C700B1103EB53A8
+:1001A000000F6003057D0965377E0A62357E7C9805
+:1001B0000A62327E0965307E12051205AD026007C3
+:1001C000027C065A8498265A277F011F0320004884
+:1001D000E87C700B11031353A5981500047809628D
+:1001E000065A0962265AA4981500150004780A6276
+:1001F000065A0A62265AA4981500150015000478BC
+:100200000B62065A0B62265A077C0000EB55004D24
+:10021000067DFAC1E3575F9807000C6813C20AC253
+:100220005C98700B110313536C07017CD9C1FB5E02
+:100230008A066B07017CD9C1F35EDB59D3588F0165
+:1002400010010F398B003CC12B7DC05AC85B4EC1D9
+:10025000277C88038906E35CFF0D1105FF1DBC05A3
+:100260003E07004D187D700811007E07097D7D074F
+:10027000027D2852DC98F852DB54BC02CC02097C87
+:100280007C07027D2852E598F852D354BC02CC0278
+:10029000097D0004D3988B00C052C85359C1D67D44
+:1002A0000002C398FF08BF007F07157D8804D500B2
+:1002B000017D8D00A005EB5D8F0212021202FF3A54
+:1002C000DA05027C3E071099A402DD02027D3E079A
+:1002D00010995E071099EB55004D027C9805EB5D77
+:1002E000F352FB546A07387D6C07017D5F996B07F9
+:1002F0007D7C6907047D6807027D010E279993586C
+:10030000D600017D8E009355A005935DA0066D0774
+:10031000067D6E070B7D02780255045D3C99150041
+:1003200004780255045D2255045D3C99150004785B
+:100330000255045D0255245D1D7C004E087C690752
+:10034000037D0255177E4699045D147F89069350FC
+:100350000048017D1F99D299150006780255045D69
+:100360004F070255245D2F07017CD29917006F07B4
+:10037000017C012093559D000700D9D9EB986907AE
+:10038000047D6807027D010E6D999358D600017DAA
+:100390008E009355A005935DA0066D07077D6E073F
+:1003A0000D7DD36C02780255C86D85991500D36C0C
+:1003B00004780255C86D2255C86D859915000578D9
+:1003C000D76C0255C86D0255C86D0F7C004E087C75
+:1003D0006907037D0255097E8F99C86D067F8906DE
+:1003E00093500048017D6599D299CC996907047DA5
+:1003F0006807027D010EA4999358D600017D8E00F6
+:100400009355A005935DA0066D07077D6E070D7DD2
+:10041000C36A0278C865045DBC9915000578C76A8F
+:10042000C865045DC865045DBC991500C36A04789D
+:10043000C865045DC865245D0F7C004E087C6907B3
+:10044000037DC865097EC699045D067F89069350C1
+:100450000048017D9C99D29993559D000700FF6C3F
+:10046000D9D9EB980000E354EB55004D017CEB9893
+:10047000D398E354EB55FF0A1102FF1A7F07027C61
+:10048000A005E6999D008C05BA05A0051002BA04E6
+:10049000AD0454040600D9C1E3C1DB57F352056A29
+:1004A0008F00D500017D8D00A005FB567804037DEB
+:1004B0007904297D1F7C79042E7CE35D700D110584
+:1004C000ED55000F6007027D0652079A2652337ED3
+:1004D0006005027D10021202096A2D7F1202096A6C
+:1004E0002A7F1202096A277F011F03200048EA7C45
+:1004F000E355329A150015001500047806520B6A70
+:1005000026520B6A319A15001500047806520A6AC1
+:1005100026520A6A319A150004780652096A265250
+:10052000096A097C286A077F0000DB57004D057DBA
+:10053000FAC1DB57F09977C254040AC2ED99E3C1BE
+:10054000DB57F352056AFB568E02941AC36AC862DF
+:100550006902267D941EC36ED36EC8624802C86AC3
+:100560009426981EC36EC8629826C36E60020E7DE4
+:10057000981EC36EC8626C02037D9826C36E809A73
+:100580004C02D36EC86A9826C36E909AC8626E02F7
+:10059000017C8A9A096A1A7F0125004D217D519AB2
+:1005A000E36E8F00D805017D8D00C8626E02127D5A
+:1005B000096A0C7F01250120F87CDB57004D107D76
+:1005C000286A047F0000FAC1DB57439A07000462DF
+:1005D0000C6A8D9A286AFA7F04627AC2580454041D
+:1005E000FF081100FF18BC00CD00017C809A286A2A
+:1005F000ED7F04627AC20AC2409AD9C1E3C1DB57D7
+:10060000F352056AFB568E02941A025269021D7D4E
+:10061000941E06524802065A9426981E06524C0210
+:10062000065A9826981E065260020A7C98260652A0
+:100630006E02237D096A1D7F0125004D247DB29A3B
+:10064000286A177F04627AC2E39A8F00D805017D79
+:100650008D00A00506526E02107D096A0A7F0120F6
+:10066000F97C286A067F0000004D0D7DFAC1DB573A
+:10067000A29A070004620C6AE09A286AFA7F046270
+:100680007AC258045404286AF47F0AC29F9A2BDB6A
+:10069000DB57F352056AC7698F00D500017D8D00D5
+:1006A000A0057804037D79041C7D157C79041E7CEB
+:1006B000EE56C862287E6006027D10021202096AA8
+:1006C000227F1202096A1F7F1202096A1C7F03201F
+:1006D0000048EF7C1A9B150015000278C8620B6A6F
+:1006E000199B15000278C8620A6A199B0278C862D1
+:1006F000096A097C286A077F004D077DD3520108EB
+:1007000002580004EB9A77C25404D35201080258ED
+:10071000010443DBE89A700B11031353DB5F0A07F4
+:10072000D3588B00FB5E3CC1157DC05AC85B4EC1DF
+:10073000117C8803F05DFF0D1105FF1DBC05004D08
+:10074000047D429B0807DB5F0A078B00C352CB5333
+:1007500059C1EC7D0002339B016E0B612F7E0B6251
+:100760002D7E0B632B7E0C0D1704170417049D04BC
+:10077000081DCC05017C0C0DD16A000F4207C86F23
+:10078000DD6F1C7F8E009D0001680B67177ED56BA7
+:1007900004080278C86F1207117C0B670F7E0408EB
+:1007A0000278C86F12070A7CDD6F087FD169010FDC
+:1007B000C86FDD6F037F01010004589B0700FF68CD
+:1007C0000C680002589BC8DBDB57F352056AFB52EA
+:1007D000C76A8F00D500017D8D00A0057804037DD8
+:1007E00079041C7D157C79041E7CEE56C862287E37
+:1007F0006006027D10021202096A227F1202096A53
+:100800001F7F1202096A1C7F03200048EF7CB79B00
+:10081000150015000278C8620B6AB69B15000278B5
+:10082000C8620A6AB69B0278C862096A097C286AAB
+:10083000077F004D077DD352010802580004879BB3
+:1008400077C25404D352010802580104E0DB849BB0
+:10085000700B11031353DB5F0A07D3588B00FB5E49
+:100860003CC1157DC05AC85B4EC1117C8803F05D48
+:10087000FF0D1105FF1DBC05004D047DDF9B080722
+:10088000DB5F0A078B00C352CB5359C1EC7D0002DA
+:02089000D09BFB
+:00000001FF
diff --git a/firmware/imx/sdma/sdma-imx7d.bin.ihex b/firmware/imx/sdma/sdma-imx7d.bin.ihex
new file mode 100644
index 0000000..b76eaef
--- /dev/null
+++ b/firmware/imx/sdma/sdma-imx7d.bin.ihex
@@ -0,0 +1,145 @@
+:1000000053444D4104000000010000001C000000AA
+:100010002A000000C400000038080000840200002C
+:10002000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFE0
+:10003000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFD0
+:10004000FFFFFFFFFFFFFFFF051A0000FFFFFFFF9D
+:10005000ED020000B1180000FFFFFFFF631A00006F
+:10006000FFFFFFFFC2030000FFFFFFFFFFFFFFFFD7
+:10007000FFFFFFFFAD020000FFFFFFFF7D03000059
+:10008000FFFFFFFFFFFFFFFF4E04000070040000B2
+:10009000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF70
+:1000A000FFFFFFFF0018000058180000B119000002
+:1000B000121B0000AD1A0000491B000000180000D0
+:1000C000FFFFFFFFE5C1DB57F352016AFB52D36A22
+:1000D000FB521C1AC36AE8621102FF3A3008D002D0
+:1000E000337C8F00D500017D8D00A005EB5D780489
+:1000F000037D79041C7D207C7904157CEE56600616
+:10010000057D0965317E0A622F7E28980A622C7E61
+:1001100009652A7E12051205AD02C86A257F0320F3
+:100120000048ED7C409802780962C86A3F98150043
+:1001300002780A62C86A3F981500150002780B62BF
+:10014000C86A127CDF6D107FFB52041AC36AD36A3F
+:10015000C862042AC86AFB52D36A0000EB55004DFE
+:10016000077DFCC1DB5706980700CC680C6815C2F8
+:100170000CC20198D9C1E5C1DB57E35FE357F352E5
+:10018000216A8F00D500017D8D00A005EB5DFB5637
+:100190007804037D79042A7D317C7904207C700BFE
+:1001A0001103EB53000F6003057D0965377E0A627A
+:1001B000357E7C980A62327E0965307E1205120512
+:1001C000AD026007027C065A8498265A277F011FD9
+:1001D00003200048E87C700B11031353A598150009
+:1001E00004780962065A0962265AA4981500150077
+:1001F00004780A62065A0A62265AA4981500150065
+:10020000150004780B62065A0B62265A077C000020
+:10021000EB55004D067DFCC1E3575F9807000C6865
+:1002200015C20CC25C98700B110313536C07017C50
+:10023000D9C1FB5E8A066B07017CD9C1F35EDB592D
+:10024000D3588F0110010F398B003CC12B7DC05A50
+:10025000C85B4EC1277C88038906E35CFF0D11054E
+:10026000FF1DBC053E07004D187D700811007E077C
+:10027000097D7D07027D2852DC98F852DB54BC02D0
+:10028000CC02097C7C07027D2852E598F852D354B1
+:10029000BC02CC02097D0004D3988B00C052C85325
+:1002A00059C1D67D0002C398FF08BF007F07157DA6
+:1002B0008804D500017D8D00A005EB5D8F02120240
+:1002C0001202FF3ADA05027C3E071099A402DD0211
+:1002D000027D3E0710995E071099EB55004D027C98
+:1002E0009805EB5DF352FB546A07267D6C07017D90
+:1002F0004D996B07577C6907047D6807027D010EE5
+:1003000027999358D600017D8E009355A005935DE3
+:10031000A00602780255045D1D7C004E087C69072A
+:10032000037D0255177E3499045D147F890693502E
+:100330000048017D1F999899150006780255045DC3
+:100340004F070255245D2F07017C989917006F070E
+:10035000017C012093559D0007009FD9EB98D36C39
+:100360006907047D6807027D010E5C999358D600E9
+:10037000017D8E009355A005935DA006027802557D
+:10038000C86D0F7C004E087C6907037D0255097E0D
+:100390006999C86D067F890693500048017D54997C
+:1003A00098999299C36A6907047D6807027D010ED6
+:1003B0007F999358D600017D8E009355A005935DDB
+:1003C000A0060278C865045D0F7C004E087C6907B2
+:1003D000037DC865097E8C99045D067F890693506C
+:1003E0000048017D7799989993559D000700FF6C0F
+:1003F0009FD9EB980000E354EB55004D017CEB983E
+:10040000D398E354EB55FF0A1102FF1A7F07027CD1
+:10041000A005AC999D008C05BA05A0051002BA0490
+:10042000AD0454040600D9C1E5C1DB57F352056A97
+:100430008F00D500017D8D00A005FB567804037D5B
+:100440007904297D1F7C79042E7CE35D700D1105F4
+:10045000ED55000F6007027D0652CD992652337E7E
+:100460006005027D10021202096A2D7F1202096ADC
+:100470002A7F1202096A277F011F03200048EA7CB5
+:10048000E355F899150015001500047806520B6A1B
+:1004900026520B6AF79915001500047806520A6A6D
+:1004A00026520A6AF799150004780652096A2652FC
+:1004B000096A097C286A077F0000DB57004D057D2B
+:1004C000FCC1DB57B69979C254040CC2B399E5C19B
+:1004D000DB57F352056AFB568E02941AC36AC86250
+:1004E0006902267D941EC36ED36EC8624802C86A34
+:1004F0009426981EC36EC8629826C36E60020E7D55
+:10050000981EC36EC8626C02037D9826C36E469A1D
+:100510004C02D36EC86A9826C36E569AC8626E02A1
+:10052000017C509A096A1A7F0125004D217D179A96
+:10053000E36E8F00D805017D8D00C8626E02127DCA
+:10054000096A0C7F01250120F87CDB57004D107DE6
+:10055000286A047F0000FCC1DB57099A0700046287
+:100560000C6A539A286AFA7F04627CC258045404C5
+:10057000FF081100FF18BC00CD00017C469A286AD4
+:10058000ED7F04627CC20CC2069AD9C1E5C1DB577B
+:10059000F352056AFB568E02941A025269021D7DBF
+:1005A000941E06524802065A9426981E06524C0281
+:1005B000065A9826981E065260020A7C9826065211
+:1005C0006E02237D096A1D7F0125004D247D789AE6
+:1005D000286A177F04627CC2A99A8F00D805017D22
+:1005E0008D00A00506526E02107D096A0A7F012067
+:1005F000F97C286A067F0000004D0D7DFCC1DB57A9
+:10060000689A070004620C6AA69A286AFA7F046254
+:100610007CC258045404286AF47F0CC2659AF1DA4B
+:10062000DB57F352056AC7698F00D500017D8D0045
+:10063000A0057804037D79041C7D157C79041E7C5B
+:10064000EE56C862287E6006027D10021202096A18
+:10065000227F1202096A1F7F1202096A1C7F03208F
+:100660000048EF7CE09A150015000278C8620B6A1A
+:10067000DF9A15000278C8620A6ADF9A0278C862B7
+:10068000096A097C286A077F004D077DD35201085B
+:1006900002580004B19A79C25404D3520108025896
+:1006A000010409DBAE9A700B11031353DB5F0A07D9
+:1006B000D3588B00FB5E3CC1157DC05AC85B4EC150
+:1006C000117C8803F05DFF0D1105FF1DBC05004D79
+:1006D000047D089B0807DB5F0A078B00C352CB53DE
+:1006E00059C1EC7D0002F99A016E0B612F7E0B62FD
+:1006F0002D7E0B632B7E0C0D1704170417049D042D
+:10070000081DCC05017C0C0DD16A000F4207C86F93
+:10071000DD6F1C7F8E009D0001680B67177ED56B17
+:1007200004080278C86F1207117C0B670F7E04085B
+:100730000278C86F12070A7CDD6F087FD169010F4C
+:10074000C86FDD6F037F010100041E9B0700FF6877
+:100750000C6800021E9B8EDBDB57F352056AFB52CE
+:10076000C76A8F00D500017D8D00A0057804037D48
+:1007700079041C7D157C79041E7CEE56C862287EA7
+:100780006006027D10021202096A227F1202096AC3
+:100790001F7F1202096A1C7F03200048EF7C7D9BAB
+:1007A000150015000278C8620B6A7C9B1500027860
+:1007B000C8620A6A7C9B0278C862096A097C286A56
+:1007C000077F004D077DD3520108025800044D9B5E
+:1007D00079C25404D352010802580104A6DB4A9B93
+:1007E000700B11031353DB5F0A07D3588B00FB5EBA
+:1007F0003CC1157DC05AC85B4EC1117C8803F05DB9
+:10080000FF0D1105FF1DBC05004D047DA59B0807CC
+:10081000DB5F0A078B00C352CB5359C1EC7D00024A
+:10082000969BFBDBDB57F352016AFB52D36A8F00C6
+:10083000D500017D8D00A005EB5D7804037D790472
+:100840001C7D207C7904157CEE566006057D0965CB
+:100850002A7E0A62287ECE9B0A62257E0965237E57
+:1008600012051205AD02C86A1E7F03200048ED7C08
+:10087000E69B02780962C86AE59B150002780A6265
+:10088000C86AE59B1500150002780B62C86A0B7CEC
+:10089000DF6D097F0000EB55004D097DD352010843
+:1008A00002580004B59B0700CC680C6815C2D352EF
+:1008B00001080258010413DCB09B700B11031353A1
+:1008C000DB5F0A07D3588B00FB5E3CC1157DC05A25
+:1008D000C85B4EC1117C8803F05DFF0D1105FF1D43
+:1008E000BC05004D047D129C0807DB5F0A078B00E6
+:0C08F000C352CB5359C1EC7D0002039CA5
+:00000001FF
-- 
1.9.1

